File: Algorithms/Scala/RandomSearch/src/test/scala/OSOL/Extremum/Algorithms/Scala/RandomSearchSuite.scala
package OSOL.Extremum.Algorithms.Scala

import OSOL.Extremum.Cores.JVM.Optimization.Testing.RealTester
import org.scalatest.FunSuite

class RandomSearchSuite extends FunSuite {

  val r: Double = 1.0
  val oneMin: Double = 60.0

  test("Warm Up")
  {
    val tester = new RealTester
    try {
      val result = tester(
        RandomSearch.createFixedStepRandomSearch(radius = 1.0 * r, maxTime = 1 * oneMin),
        RandomSearch.createFixedStepRandomSearch(radius = 0.5 * r, maxTime = 2 * oneMin),
        RandomSearch.createFixedStepRandomSearch(radius = 0.1 * r, maxTime = 5 * oneMin))
    }
    finally {
      assert(true)
    }
  }

  test("Test Random Search")
  {
    val tester = new RealTester
    assert(tester(
      RandomSearch.createFixedStepRandomSearch(radius = 1.0 * r, maxTime = 1 * oneMin),
      RandomSearch.createFixedStepRandomSearch(radius = 0.5 * r, maxTime = 2 * oneMin),
      RandomSearch.createFixedStepRandomSearch(radius = 0.1 * r, maxTime = 5 * oneMin)))
  }

}

File: Algorithms/Java/RandomSearch/src/test/java/OSOL/Extremum/Algorithms/Java/RandomSearchSuite.java
package OSOL.Extremum.Algorithms.Java;

import OSOL.Extremum.Cores.JVM.Optimization.Algorithm;
import OSOL.Extremum.Cores.JVM.Optimization.Testing.RealTester;
import OSOL.Extremum.Cores.JVM.Vectors.RealVector;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.*;
import scala.collection.JavaConverters;


public class RandomSearchSuite {

    double r = 1.0;
    double oneMin = 60.0;

    @Test
    void TestRandomSearch()
    {
        RealTester tester = new RealTester();
        List<Algorithm<RealVector, Double, RealVector>> configsToTest = new ArrayList<>();
        configsToTest.add(RandomSearch.createFixedStepRandomSearch(1.0 * r, 1 * oneMin));
        configsToTest.add(RandomSearch.createFixedStepRandomSearch(0.5 * r, 2 * oneMin));
        configsToTest.add(RandomSearch.createFixedStepRandomSearch(0.1 * r, 5 * oneMin));

        assertTrue(tester.apply(JavaConverters.iterableAsScalaIterable(configsToTest).toSeq()));
    }
}
